tests/unit/Core/Content/Media/Infrastructure/Path/MediaUrlGeneratorTest.php

```diff
@@ -62,5 +82,37 @@ public static function generateProvider(): \Generator
             new UrlParams('id', UrlParamsSource::MEDIA, 'https://test.com/photo/flower.jpg', null),
             'https://test.com/photo/flower.jpg',
         ];
+
+        yield 'Test with whitespaces in path' => [
+            new UrlParams('id', UrlParamsSource::MEDIA, 'media/foo/3a/test file.jpg', null),
+            Feature::isActive('v6.8.0.0')
+                ? 'http://localhost:8000/media/foo/3a/test%20file.jpg'
+                : 'http://localhost:8000/media/foo/3a/test file.jpg',
```

Comment:
Like you already do here, so I don't think all the magic is needed or did I miss something? 🙈